### PR TITLE
draft: feature: add bookworm 12

### PR DESCRIPTION
**Pull request by maxjuhlke**
This feature will try to enable distroless 12.

This tries to re-implement:

https://github.com/GoogleContainerTools/distroless/pull/1146

### CODE REVIEW

**Comment by evanj** [comment]
Hey nice! I did some of the initial work for Debian 11 support in Distroless, so I just was checking in on this project to see if any progress has been made to add Debian 12 support. I'm glad to see there is interest in pushing this forward! I am not a maintainer, but I'll try to find some time to look at this PR. Thanks for doing this!

File: debian_package_manager/internal/build/config/distros.go [hunk 1]
```diff
@@ -54,6 +59,8 @@ func (d Distro) Codename() string {
 		return "buster"
 	case DEBIAN11:
 		return "bullseye"
+	case DEBIAN12:
+		return "bullseye"
```
Comment (tomtwinkle) [inline debian_package_manager/internal/build/config/distros.go:63]: bookworm?

File: BUILD [hunk 2]
```diff
@@ -132,6 +135,8 @@ CC_VARIANTS = [
 CC = {
     "{REGISTRY}/{PROJECT_ID}/cc:{COMMIT_SHA}": "//cc:cc_root_amd64_debian11",
     "{REGISTRY}/{PROJECT_ID}/cc-debian11:{COMMIT_SHA}": "//cc:cc_root_amd64_debian11",
+    "{REGISTRY}/{PROJECT_ID}/cc-debian12:{COMMIT_SHA}": "//cc:cc_root_amd64_debian12",
+
```
Comment (n0npax) [inline BUILD:139]: I guess that this new line line will be breaking lint

File: base/testdata/debian12.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+schemaVersion: "1.0.0"
+fileContentTests:
+- name: 'os-release contents'
+  path: '/etc/os-release'
+  expectedContents: ['.*\nVERSION="Debian GNU/Linux 12 \(bookworm\)"\n']
```
Comment (n0npax) [inline base/testdata/debian12.yaml:5]: should new line at end of file be added?

File: debian_packages.yaml [hunk 4]
```diff
@@ -1,12 +1,11 @@
-# debian 11
-- distros: ["debian11"]
-  archs: ["amd64", "arm64", "arm", "s390x", "ppc64le"]
```
Comment (n0npax) [inline debian_packages.yaml:3]: This doesn't seem right.

Plenty of folks us using `arm64` cpu. Also removing less common stuff like `power pc` without wide discussion seems to be wrong in my opinion. I'd bet maintainers would reject PR base on this change

**Comment by n0npax** [comment]
looks related to: https://github.com/GoogleContainerTools/distroless/issues/1337

**Comment by loosebazooka** [comment]
thanks for working on this, I think we have this actually completed in #1385

**Pull request closed without merging**